### PR TITLE
Add Looper

### PR DESCRIPTION
**Pull request by shriramters**

### CODE REVIEW

**Comment by shriramters** [comment]
# Changes
- Add looper
  - Recording
  - Overdubbing 
- Handle Control Change, Program Change, Pitch Wheel midi messages
- Add General MIDI 1 Instrument selection (aka program change) : `Preferences` -> `Piano` -> `Program Change`
- Add new font Metropolis (Patched with CC0 flat, sharp and natural symbols)
- Add backend switcher `Preferences`->`Configuration`->`Backend`

File: src/gm_instrument_table.hpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+// Reinvent JSON
+inline void showGMInstrumentTable() {
+    struct Instrument {
+        int code;
+        const char* name;
+    };
+
+    struct Category {
+        const char* name;
+        Instrument instruments[16];
```
Comment (akashters) [inline src/gm_instrument_table.hpp]: vector size can be 8 and give a name to this magic number

File: src/looper.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#include "looper.hpp"
+#include "midi_event.hpp"
+#include <iostream>
```
Comment (akashters) [inline src/looper.cpp]: not required

File: src/piano.hpp [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
+    void keyOff(int midi_note_number, int chan);
+    void keyToggle(int midi_note_number);
     void clearAllKeys();
     std::vector<size_t> getPressedNotes();
     fluid_synth_t* getSynth();
     void processEvent(sf::Event& event);
+    void midiEvent(const MidiEvent& me);
+    void setMidiEventCallback(MidiEventCallback callback);
     float key_aspect_ratio = 4.f;
+    int getChannel();
```
Comment (akashters) [inline src/piano.hpp]: mark method as const

File: src/preferences.hpp [hunk 4]
```diff
@@ -18,7 +19,7 @@ struct Preferences {
         std::array<float, 4> pressed_note_colors;
         Piano() {
             gain = 2.f;
```
Comment (akashters) [inline src/preferences.hpp]: default gain to 1.0

**Review by akashters — CHANGES_REQUESTED** [review]

**Review by akashters — APPROVED** [review]

**Pull request merged**